vernac/vernacstate.mli

```diff
@@ -50,8 +50,6 @@ type t =
   (** summary + libstack *)
   ; lemmas  : LemmaStack.t option
   (** proofs of lemmas currently opened *)
-  ; program : Declare.OblState.t NeList.t
-  (** program mode table. One per open module/section including the toplevel module. *)
```

Comment:
Following discussion on the alternative PR #15729 , this bit here is quite useful to document managers, for example to provide a richer goal and obligations list.